### PR TITLE
Changed Font to Inter and added Typescript

### PR DESCRIPTION
**Pull request by wbridegroom**
Gitlab Issue: https://gitlab.mx.com/mx/experiences/money-experience-web/mew/issues/-/issues/520

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -73,12 +91,12 @@ export default {
...
     TransparentScrim: 'rgba(0, 0, 0, 0.25)',
     TransparentScrimDarker: 'rgba(0, 0, 0, 0.50)',
-  },
+  } as ColorType,
   Font: {
-    Regular: 'ProximaNovaRegular, Helvetica, Arial, sans-serif',
-    Semibold: 'ProximaNovaSemibold, Helvetica, Arial, sans-serif',
-    Bold: 'ProximaNovaBold, Helvetica, Arial, sans-serif',
-  },
+    Regular: 'Inter, Helvetica, Arial, sans-serif',
```
Comment (jgerlach) [inline src/core.ts]: Do we want to match the old naming pattern so this is `InterRegular`, `InterSemibold`, `InterBold`?

File: src/core.ts [hunk 2]
```diff
@@ -87,43 +105,43 @@ export default {
...
+    H1: 31,
+  } as FontSizeType,
   FontWeight: {
     Regular: 400,
     Semibold: 600,
     Bold: 700,
-  },
+  } as FontWeightType,
   LetterSpacing: {
     Uppercase: 0.48,
```
Comment (jgerlach) [inline src/core.ts:132]: I'm wondering if we should add the new letter spacing values here instead of just in the letterSpacing file.

Comment (jgerlach) [reply src/core.ts:132]: Looks like you added them in the `TokenType` file, but not here.

File: src/TokenType.ts [hunk 3]
```diff
@@ -0,0 +1,475 @@
+export interface BackgroundColorType {
```
Comment (jgerlach) [inline src/TokenType.ts]: NIT: I think it would make more sense to call this file `TokenTypes.ts`

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  NATIVE: 'native',
+}
+
+export const themes = {
+  LIGHT: 'light',
+  DARK: 'dark',
+}
+
+const getSectionTokens = (theme = themes.LIGHT, base = core) => ({
+  BackgroundColor: theme === themes.LIGHT ? backgroundColor.light(base) : backgroundColor.dark(base),
```
Comment (jgerlach) [inline src/index.ts:27]: Why did you change these to a ternary? Is that because of the typescript change?

File: src/mods.ts [hunk 5]
```diff
@@ -1,15 +1,10 @@
...
 
 // Shared modifications
 const updatedBoxShadow = updateValues(core.BoxShadow, '')
-const updatedEasing = updateValues(core.Easing, {
-  x1: 0.475,
-  y1: 0.425,
-  x2: 0,
-  y2: 0.995,
-})
+const updatedEasing = updateValues(core.Easing, 'cubic-bezier(0.475, 0.425, 0, 0.995)')
```
Comment (jgerlach) [inline src/mods.ts:7]: Is this different then the code on the left? I didn't write it, so I'm actually not sure what it is doing.

File: src/tokens/fontFamily.js [hunk 6]
```diff
@@ -1,14 +0,0 @@
-const light = (core) => ({
```
Comment (jgerlach) [inline src/tokens/fontFamily.js:1]: It does look like we aren't using this anywhere. Nice catch.

File: src/tokens/letterSpacing.js [hunk 7]
```diff
@@ -1,14 +0,0 @@
-const light = (core) => ({
```
Comment (jgerlach) [inline src/tokens/letterSpacing.js:1]: It looks like the other files were renamed, but this one is showing as deleted. Is that just a git code diff bug?

File: src/utils.ts [hunk 8]
```diff
@@ -98,15 +98,15 @@ export const splitTokens = (tokens, core, leaf) => {
 
   const tokenList = expandTokens(tokens)
   tokenList.map((token) => {
-    const type = getTokenType(token, core, leaf)
+    const tokenType = getTokenType(token, core, leaf)
```
Comment (jgerlach) [inline src/utils.ts:101]: I wonder if now that we have types for the tokens if this function could be confusing since it's not actually getting TypeScript types

File: src/core.ts (hunk 1 above)
Comment (wbridegroom) [reply src/core.ts]: I was originally planning to get this route, but it looks like `Inter` doesn't use different names like that. It's simply `Inter` with different weights. Unless I'm missing something.

File: src/index.ts (hunk 4 above)
Comment (wbridegroom) [reply src/index.ts:27]: Yes, this is a Typescript issue. It doesn't allow indexing like that.

File: src/mods.ts (hunk 5 above)
Comment (wbridegroom) [reply src/mods.ts:7]: TBH, I'm also not sure what the purpose of this is, because the values provided are identical to the default values.  This was the only way I could get it to work. Out of all the changes I made this is the only one I'm not 100% certain about.

File: src/utils.ts (hunk 8 above)
Comment (wbridegroom) [reply src/utils.ts:101]: I agree, but I was trying to avoid making too many changes. Also, I couldn't think of a better name.

File: src/core.ts [hunk 9]
```diff
@@ -73,12 +91,12 @@ export default {
...
-  },
+  } as ColorType,
   Font: {
-    Regular: 'ProximaNovaRegular, Helvetica, Arial, sans-serif',
-    Semibold: 'ProximaNovaSemibold, Helvetica, Arial, sans-serif',
-    Bold: 'ProximaNovaBold, Helvetica, Arial, sans-serif',
-  },
+    Regular: 'Inter, Helvetica, Arial, sans-serif',
+    Semibold: 'Inter, Helvetica, Arial, sans-serif',
+    Bold: 'Inter, Helvetica, Arial, sans-serif',
```
Comment (smstamm) [inline src/core.ts]: If `Semibold` and `Bold` will be the same as `Regular`, should we just remove these as options and allow them to be controlled in `FontWeight`?

Comment (wbridegroom) [reply src/core.ts]: Long term yes, however, in order to keep this backwards compatible, I think we need to leave it for now.

File: src/core.ts (hunk 1 above)
Comment (jgerlach) [reply src/core.ts]: So, the name itself is actually dependent on how you import the font into your app. Here is an example of how we are currently importing fonts

```
@font-face {
  font-family: 'ProximaNovaSemibold';
  src: url('https://s3.amazonaws.com/path-to-font/fonts/proxima-nova/semibold.eot');
  font-weight: 600;
  font-style: normal;
}
```

You can see that `font-family` is specified as `ProximaNovaSemibold`, but the font file itself isn't named that.

File: src/tokens/letterSpacing.js (hunk 7 above)
Comment (wbridegroom) [reply src/tokens/letterSpacing.js:1]: I think this was just a diff bug, but I'm removing the TS version anyway now since it's no longer needed.

File: src/core.ts (hunk 1 above)
Comment (wbridegroom) [reply src/core.ts]: Changed Inter to match the old naming pattern.

File: src/core.ts (hunk 2 above)
Comment (wbridegroom) [reply src/core.ts:132]: Merged into core and removed the separate file.

File: src/TokenType.ts (hunk 3 above)
Comment (wbridegroom) [reply src/TokenType.ts]: Done

File: src/core.ts (hunk 9 above)
Comment (wbridegroom) [reply src/core.ts]: Changed Inter to match the old naming pattern, so now we have `InterRegular`, `InterSemiBold` and `InterBold`

**Review by smstamm — APPROVED** [review]
It looks like you addressed all feedback. Looks good to me! Nice work!

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ---
 
+## [11.0.0] - 08-04-2023
```
Comment (jgerlach) [inline CHANGELOG.md:10]: Thanks for updating this.

**Review by jgerlach — APPROVED** [review]
Great job with the updates and moving to TypeScript

File: src/mods.ts (hunk 5 above)
Comment (jgerlach) [reply src/mods.ts:7]: If we notice any issues after we publish we will have to figure out how to fix it.
